rotkehlchen/exchanges/bitstamp.py

```diff
@@ -0,0 +1,711 @@
+        try:
+            base_asset = Asset(base_asset_symbol)
+            quote_asset = Asset(quote_asset_symbol)
+        except UnknownAsset as e:
+            log.error(str(e))
+            raise DeserializationError(
+                f'Unknown asset {e.asset_name} found while processing trade pair.',
+            )
+
+        return TradePairData(
+            pair=pair,
+            base_asset_symbol=base_asset_symbol,
+            quote_asset_symbol=quote_asset_symbol,
+            base_asset=base_asset,
+            quote_asset=quote_asset,
+            trade_pair=TradePair(f'{base_asset.identifier}_{quote_asset.identifier}'),
+        )
+
+    @overload  # noqa: F811
+    def _process_unsuccessful_response(  # pylint: disable=no-self-use
+            self,
+            response: Response,
+            case: Literal['validate_api_key'],
+    ) -> Tuple[bool, str]:
+        ...
+
+    @overload  # noqa: F811
+    def _process_unsuccessful_response(  # pylint: disable=no-self-use
+            self,
+            response: Response,
+            case: Literal['balances'],
+    ) -> Tuple[Optional[Dict[Asset, Balance]], str]:
+        ...
+
+    @overload  # noqa: F811
+    def _process_unsuccessful_response(  # pylint: disable=no-self-use
+            self,
+            response: Response,
+            case: Literal['trades'],
+    ) -> List[Trade]:
+        ...
+
+    @overload  # noqa: F811
+    def _process_unsuccessful_response(  # pylint: disable=no-self-use
+            self,
+            response: Response,
+            case: Literal['asset_movements'],
+    ) -> List[AssetMovement]:
+        ...
+
+    def _process_unsuccessful_response(
+            self,
+            response: Response,
+            case: str,
+    ) -> Union[
+        List[Trade],
+        List[AssetMovement],
+        Tuple[bool, str],
+        Tuple[Optional[Dict[Asset, Balance]], str],
+    ]:
+        """This function processes not successful responses for the following
+        cases listed in `case`.
+        """
+        case_pretty = case.replace('_', ' ')  # human readable case
+        # Process error response as a dict
+        try:
+            response_dict = rlk_jsonloads_dict(response.text)
+        except JSONDecodeError:
+            msg = f'Bitstamp returned invalid JSON response: {response.text}.'
+            log.error(msg)
+
+            # Custom response per case
+            if case in {'validate_api_key', 'balances'}:
+                raise RemoteError(msg)
+            elif case in {'trades', 'asset_movements'}:
+                self.msg_aggregator.add_error(
+                    f'Got remote error while querying Bistamp {case_pretty}: {msg}',
+                )
+                no_results: Union[List[Trade], List[AssetMovement]] = []  # type: ignore
+                return no_results
+            else:
+                raise AssertionError(f'Unexpected Bitstamp response_case: {case}.')
+
+        # Get error code
```

Comment:
A request:

1. Please don't add unnecessary whitespace. There is no need for whitespace before and after a function call.
2. Don't add unneded comments:
    - Get error code is not needed. It's obvious you are doing it.
    - Local system clock not synch not needed. The function is named perfectly well.